eng/common/pipelines/templates/steps/verify-links.yml

```diff
@@ -8,10 +8,12 @@ parameters:
   Urls: '(Get-ChildItem -Path ./ -Recurse -Include *.md)'
   BranchReplaceRegex: "^(${env:SYSTEM_PULLREQUEST_SOURCEREPOSITORYURI}.*/(?:blob|tree)/)master(/.*)$"
   BranchReplacementName: "${env:SYSTEM_PULLREQUEST_SOURCECOMMITID}"
+  RunOnFailure: $false
 
 steps:
   - task: PowerShell@2
     displayName: Link verification check
+    condition: or(and(succeededOrFailed(), '${{ parameters.RunOnFailure }}'), and(succeeded(), eq('${{ parameters.RunOnFailure }}', false)))
```

Comment:
If my original suggestion didn't work I'd recommend we go back to your original change ad set the condition as the parameter itself with a default of `succeeded` and someone can pass in a custom condition if they want.

Reply:
Then people has to pass the `succeeded()` on top level instead of `$true` or `$false`.

Reply:
The if clause on `condition` only does not work as expected, it introduced format error.

Reply:
You can default it to that value and then if they want to run it on failure as well then the pass in `succeededOrFailed()` as the condition.

Reply:
Got you. Sure. Will have comments on how to set the parameter` condition`.